pylint/checkers/classes/class_checker.py

```diff
@@ -2240,15 +2240,17 @@ def _is_mandatory_method_param(self, node: nodes.NodeNG) -> bool:
 
 
 def _ancestors_to_call(
-    klass_node: nodes.ClassDef, method: str = "__init__"
+    klass_node: nodes.ClassDef, method_name: str = "__init__"
 ) -> dict[nodes.ClassDef, bases.UnboundMethod]:
     """Return a dictionary where keys are the list of base classes providing
     the queried method, and so that should/may be called from the method node.
     """
     to_call: dict[nodes.ClassDef, bases.UnboundMethod] = {}
     for base_node in klass_node.ancestors(recurs=False):
         try:
-            init_node: bases.UnboundMethod = next(base_node.igetattr(method))
+            init_node: bases.UnboundMethod = next(base_node.igetattr(method_name))
+            if not isinstance(init_node, astroid.UnboundMethod):
+                continue
```

Comment:
```suggestion
            init_node = next(base_node.igetattr(method_name))
            if not isinstance(init_node, astroid.UnboundMethod):
                continue
```

:smile: Apparently this typing is not correct right?

Reply:
Was thinking the same thing. Thanks for the suggestion.